src/parsing/parser.rs

```diff
@@ -3058,22 +3058,25 @@ fn handle_jsx_surrounding_parens<'a>(inner_items: PrintItems, context: &mut Cont
   let inner_items_rc = inner_items.into_rc_path();
 
   items.push_info(start_info);
-  items.push_condition(if_true_or(
+  items.push_condition(Condition::new_with_dependent_infos(
```

Comment:
This is necessary for a wrapping bug in our fork, for an input that looks like:
```tsx
func(<Foo prop1={prop1} prop2={prop2}/>);
```
and when the lineWidth needs to wrap somewhere in the props. 

1. It prints `func(<Foo prop1={prop1}` and then realises it needs to wrap
2. It reverts to the save point `func(<Foo` and then prints `prop1={prop1}` on a new line
3. At that point, this condition `parensOrNewlinesIfMultipleLines` has resolved, and since it isn't dependent on the `endInfo`, it never recomputes
4. We end up with JSX on multiple lines, but no brackets:
```tsx
func(<Foo
    prop1={prop1}
    prop2={prop2}
/>);
```

Reply:
actually, after reverting this change and testing something like:
```
expect(<SomeElement arg1={arg1} a2={a2} a3={a3} a4={a4} />).toMatchRenderedSnapshot();
```
and looking at the trace, it does the following prints:
```
expect(
····<SomeElement·arg1={arg1}·a2={a2}·a3={a3}
········a4={a4}·
```
```
expect(
····<SomeElement
········arg1={arg1}
········a2={a2}
········a3={a3}
········a4={a4}
····/>
```
```
expect(
····<SomeElement
········arg1={arg1}
········a2={a2}
········a3={a3}
········a4={a4}
····/>
```

At which point it does revert all the way back to `parensOrNewlinesIfMultipleLines`, and reprint with the bracket:
```
expect(
····(
········<SomeElement·arg1={arg1}·a2={a2}·a3={a3}
············a4={a4}
```
```
expect(
····(
········<SomeElement
············arg1={arg1}
············a2={a2}
············a3={a3}
············a4={a4}
········/>
····),
```

It looks like `look_ahead_info_save_points` is correctly populated with the right place here, so I'll look into this in more details first and see if it's something to be fixed in our fork. Since the look ahead should be automatically making `end_info` re-evaluate the condition anyway.

Reply:
Hm, weird. I reverted the change and it works with our custom configuration as well, for this test case, but it's still failing on the same structure in our repo code. I'll investigate further.

Reply:
The issue is caused by a seemingly innocuous config I added a while back. Then I looked at the code for it and found this little gem:
![image](https://user-images.githubusercontent.com/13516619/135723141-c9a20e9c-3ef0-4506-b07d-e530e4de4fee.png)

😂

Reply:
Fixed the issue. I've reverted this block now.